### DISCUSSION TITLE
Cluster Questions

### DISCUSSION THREAD

**Discussion by dmccabe79** [mKeRix/room-assistant#500] [Q&A] [answered]
If I want to add my HA OS pi4 that's currently running an instance of RA, along with 4 remote raspberry pi zero Ws... what address do I add to the cluster config? All my others are connecting via mDNS but the central add-on on the pi4 isn't.

**Comment by mKeRix** [comment]
Do you have a static IP or DHCP reservation on any of them? In that case I would just add that as `<static-ip>:6415` to at least one of your instances. On discovery the instances will share their knowledge, so in theory you don't need to configure the peers everywhere.

**Reply by dmccabe79** [reply]
I do not have static IPs setup on the 4 pi zeros. I probably just worded my question wrong. My 4 pis are connecting to mDNS perfectly. The Pi4 that's running HA OS is the one instance that can't connect to mDNS. I'm trying to add it to the cluster for a total of 5 instances.

**Comment by dmccabe79** [comment, marked as answer]
The whole thing is acting very strange. It's saying im closer to the "bathroom" when im actually in the kitchen. I feel like I'm missing something along the way here. Again, I am running HAOS on a pi4, with the add-on. I'm also running 4 pi zeros throughout. I'd like to cluster all 5 of those devices if possible.

Also, is it common for it to create a cluster for each pi zero? I have kitchen cluster, bedroom cluster, bathroom cluster...etc.

**Reply by mKeRix** [reply]
Home Assistant OS does not support mDNS fully due to technical limitations, so you will have to set a static IP to either at least one of your Pi Zeros or the Pi 4. Then just configure that static IP in at least one of the other instances. The rest will be exchanged during the initial discoveries.

For the wrong states, you will have to do some trial and error to get the right tuning. You can try playing with the `maxDistance` parameters and the hardware placement to get better results. The API should help you with that as well, as it will show you the detailed measured values in the entities endpoint.

The cluster entities will be created for each instance to grant you more visibility if something goes wrong.

**Reply by dmccabe79** [reply]
Working great now. Tweaking the maxDistance is just going to take some trial and error.
Thanks!
